src/values/ArrayValue.js

```diff
@@ -242,7 +242,7 @@ export default class ArrayValue extends ObjectValue {
   }
 
   static isIntrinsicAndHasWidenedNumericProperty(obj: Value): boolean {
-    if (obj instanceof ArrayValue && obj.intrinsicName !== undefined) {
+    if (obj instanceof ArrayValue && obj.intrinsicName !== undefined && obj.isScopedTemplate !== undefined) {
       const prop = obj.unknownProperty;
```

Comment:
Maybe add `invariant(ObjectValue.isIntrinsicDerivedObject(obj));` to keep the two functions in sync.